src/database_impl/setup.rs

```diff
@@ -0,0 +1,147 @@
+///      };
+
+///      let opts = OptsBuilder::new()
+///          .ip_or_hostname(Some(config.ip_or_host.clone()))
+///          .tcp_port(config.tcp_port)
+///          .user(Some(config.user.clone()))
+///          .pass(config.pass.clone())
+///          .db_name(Some(config.db_name.clone()));
+///      let conn = Conn::new(opts)
+///          .map_err(|e| SqlnessError::ConnFailed { source: e, config })
+///          .unwrap();
+///      let db = MysqlDatabase {
+///          conn: Arc::new(Mutex::new(conn)),
+///      };
+///      let query = "select * from test;";
+///      let future = MysqlDatabase::execute(query, Arc::clone(&db.conn));
+///      let result = tokio_test::block_on(future);
+///      println!("{}", result);
+///  }
+/// ```
+#[derive(Debug)]
+pub struct MysqlDatabase {
+    conn: Arc<Mutex<Conn>>,
+}
+
+struct MysqlFormatter {
+    pub rows: Vec<Row>,
+}
+
+#[async_trait]
+impl DatabaseBuilder for MysqlDatabaseBuilder {
+    async fn build(&self, config: DatabaseConnConfig) -> Result<Box<dyn Database>> {
+        let opts = OptsBuilder::new()
+            .ip_or_hostname(Some(config.ip_or_host.clone()))
+            .tcp_port(config.tcp_port)
+            .user(Some(config.user.clone()))
+            .pass(config.pass.clone())
+            .db_name(Some(config.db_name.clone()));
+
+        let conn = Conn::new(opts).map_err(|e| SqlnessError::ConnFailed { source: e, config })?;
+        Ok(Box::new(MysqlDatabase {
+            conn: Arc::new(Mutex::new(conn)),
+        }))
+    }
+}
+
+#[async_trait]
+impl Database for MysqlDatabase {
+    async fn query(&self, query: String) -> Box<dyn Display> {
+        Self::execute(&query, Arc::clone(&self.conn)).await
+    }
+}
+
+impl MysqlDatabase {
+    async fn execute(query: &str, connect: Arc<Mutex<Conn>>) -> Box<dyn Display> {
+        let mut conn = match connect.lock() {
+            Ok(conn) => conn,
+            Err(e) => return Box::new(format!("Failed to get connect, err: {:?}", e)),
+        };
+
+        let result = conn.query_iter(query);
+        Box::new(match result {
+            Ok(result) => {
+                let mut rows = vec![];
+                let affected_rows = result.affected_rows();
+                for row in result {
+                    match row {
+                        Ok(r) => rows.push(r),
+                        Err(e) => {
+                            return Box::new(format!("Failed to parser query result, err: {:?}", e))
+                        }
+                    }
+                }
+
+                if rows.is_empty() {
+                    format!("affected_rows: {}", affected_rows)
+                } else {
+                    format!("{}", MysqlFormatter { rows })
+                }
+            }
+            Err(e) => format!("Failed to execute query, err: {:?}", e),
+        })
+    }
+}
+
+pub async fn _build_default_database<T>(config: DatabaseConnConfig) -> Result<Box<dyn Database>>
```

Comment:
When will this method be used?

Reply:
This method is the one I used to create the default database implementation.
```rust
let mysql_db = build_default_database::<MysqlDatabaseBuilder>(config)?;
```
But right now sqlness doesn't need it anywhere

Reply:
I have on clear idea on when will this be used, so I suggest remove this method for now.